### PR TITLE
Introduce some yargs

### PR DESCRIPTION
**Pull request by silasbw**
Given the proliferation of new arguments, it would be useful to bring some
argument handling conventions to the code. The rest of our projects use yargs
for this. This isn't a full conversion to yargs, but a start.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -25,18 +26,44 @@ async function main() {
     Object.assign(process.env, await loadSecret(secretName))
   }
 
-  if (!process.env.EXCHANGE_RATES_API_ACCESS_KEY) {
-    throw new Error('Missing required EXCHANGE_RATES_API_ACCESS_KEY')
-  }
+  const argv = yargs
+    .env('')
```
Comment (silasbw) [inline src/index.ts:30]: Note: yargs parses environment variables too!  so you can either do `--exchange-rates-api-access-key=foo` or `EXCHANGE_RATES_API_ACCESS_KEY=foo`.

File: package.json [hunk 2]
```diff
@@ -38,16 +38,15 @@
     "dotenv": "^10.0.0",
     "express": "^4.17.1",
     "express-prom-bundle": "^6.3.4",
-    "knex": "^0.95.14",
```
Comment (silasbw) [inline package.json:41]: these were listed twice in package.json

File: src/index.ts [hunk 3]
```diff
@@ -25,18 +26,43 @@ async function main() {
...
+      demandOption: true,
+    })
+    .option('blockchain-db-pass', {
+      description: 'Blockchain DB pass',
+      type: 'string',
+      demandOption: true,
+    })
+    .epilogue(
+      'Always specify arguments as environment variables. Not all arguments are ' +
+        'supported as CLI ones yet.',
```
Comment (jeanregisser) [inline src/index.ts:58]: nit: ```suggestion
      'Always specify arguments as environment variables. Not all arguments are supported as CLI ones yet.',
```

File: src/index.ts (hunk 1 above)
Comment (jeanregisser) [reply src/index.ts:30]: Interesting trick, so this throws an exception when missing?

**Review by jeanregisser — APPROVED** [review]

File: src/index.ts [hunk 4]
```diff
@@ -25,18 +26,43 @@ async function main() {
...
-    !process.env.BLOCKCHAIN_DB_DATABASE ||
-    !process.env.BLOCKCHAIN_DB_USER ||
-    !process.env.BLOCKCHAIN_DB_PASS
-  ) {
-    throw new Error("Blockchain database secrets couldn't be obtained")
-  }
+async function main() {
+  const args = await parseArgs()
 
   const db = await initDatabase()
```
Comment (dievazqu) [inline src/index.ts:67]: To initialize the DB we are using `process.env.BLOCKCHAIN_DB_HOST  and others. Should we change that and get those values from `args`?

File: src/index.ts (hunk 1 above)
Comment (silasbw) [reply src/index.ts:30]: Yes

File: src/index.ts (hunk 4 above)
Comment (silasbw) [reply src/index.ts:66]: Yes we should -- those changes are more invasive so i was going to do them in self-contained PRs.  This one was simply focused on adding yargs.  The hope is that authors of other PRs could know to use yargs while I loop back around and address things like you point out.  Seem reasonable?

**Review by dievazqu — APPROVED** [review]
LGTM

File: src/index.ts (hunk 4 above)
Comment (dievazqu) [reply src/index.ts:66]: Sure!

**Pull request merged**